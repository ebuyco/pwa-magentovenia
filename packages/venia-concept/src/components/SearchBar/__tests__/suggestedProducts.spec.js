import React from 'react';
import TestRenderer from 'react-test-renderer';

import SuggestedProducts from '../suggestedProducts';

jest.mock('src/classify');
jest.mock('src/drivers', () => ({
    Link: ({ children }) => children,
    resourceUrl: jest.fn()
}));

const defaultProps = {
    handleOnProductOpen: jest.fn(),
    items: [
        {
            id: 1,
            url_key: 'urlKey',
            small_image: 'smallImg',
            name: 'Product Name',
            price: {
                regularPrice: {
                    amount: {
                        currency: 'USD',
                        value: 3.5
                    }
                }
            }
        }
    ]
};

test('renders a suggestedProduct component', () => {
    const component = TestRenderer.create(
        <SuggestedProducts {...defaultProps} />
    );

    expect(component.toJSON()).toMatchSnapshot();
});
