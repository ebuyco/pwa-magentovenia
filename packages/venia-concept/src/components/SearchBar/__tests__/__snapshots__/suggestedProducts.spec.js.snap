// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a suggestedProduct component 1`] = `
<div
  className="root"
>
  <h4
    className="title"
  >
    <span
      className="titleText"
    >
      Product Suggestions
    </span>
  </h4>
  <ul
    className="items"
  >
    <li
      className="root"
    >
      <img
        alt="Product Name"
        className="productImage"
      />
      Product Name
      <div />
    </li>
  </ul>
</div>
`;
